### PR TITLE
[Merged by Bors] - refactor(Topology/ContinuousFunction/Algebra): lattice ordered group gives inf/sup formula

### PR DESCRIPTION
**Pull request by mans0954**
Previously the following comment occured in `Topology.ContinuousFunction.Algebra`:

> -- TODO:
-- This lemma (and the next) could go all the way back in `Algebra.Order.Field`,
-- except that it is tedious to prove without tactics.
-- Rather than stranding it at some intermediate location,
-- it's here, immediately prior to the point of use.

Subsequently, the theory of lattice ordered groups has been developed in Mathlib (Algebra.Order.LatticeGroup). This now provides the natural "intermediate location" for these lemmas, they are an immediate consequence of `LatticeOrderedCommGroup.two_inf_eq_add_sub_abs_sub` and `LatticeOrderedCommGroup.two_sup_eq_add_add_abs_sub`. In fact we can show that `C(α, β)` is itself a lattice ordered group and hence expressions for the `inf` and `sup` (`inf_eq` and `sup_eq`) can be deduced directly from `LatticeOrderedCommGroup.two_inf_eq_add_sub_abs_sub` and `LatticeOrderedCommGroup.two_sup_eq_add_add_abs_sub`.

This was previously submitted to Mathlib https://github.com/leanprover-community/mathlib/pull/18780

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/LatticeGroup.lean [hunk 1]
```diff
@@ -62,9 +64,21 @@ lattice, ordered, group
...
 
-variable {α : Type u} [Lattice α] [CommGroup α]
+variable {α : Type u} {β : Type v}
+
+-- A linearly ordered additive commutative group is a lattice ordered commutative group
+
+@[to_additive] -- see Note [lower instance priority]
+instance LinearOrderedCommGroup.to_covariant_class (α : Type u)
+  [LinearOrderedCommGroup α] : CovariantClass α α (· * · ) (· ≤ ·) :=
+{ elim := fun a _ _ bc => OrderedCommGroup.mul_le_mul_left _ _ bc a }
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/LatticeGroup.lean:76]: This instance already exists more generally as `OrderedCommGroup.to_covariantClass_left_le`

File: Mathlib/Algebra/Order/LatticeGroup.lean [hunk 2]
```diff
@@ -59,12 +61,10 @@ lattices.
 lattice, ordered, group
 -/
 
-
--- Needed for squares
 -- Needed for squares
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/LatticeGroup.lean:64]: What does this comment mean?

File: Mathlib/Topology/ContinuousFunction/StoneWeierstrass.lean [hunk 3]
```diff
@@ -118,7 +118,7 @@ theorem abs_mem_subalgebra_closure (A : Subalgebra ℝ C(X, ℝ)) (f : A) :
 
 theorem inf_mem_subalgebra_closure (A : Subalgebra ℝ C(X, ℝ)) (f g : A) :
     (f : C(X, ℝ)) ⊓ (g : C(X, ℝ)) ∈ A.topologicalClosure := by
-  rw [inf_eq]
+  rw [(inf_eq_half_smul_add_sub_abs_sub' ℝ)]
```
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/StoneWeierstrass.lean:121]:
```suggestion
  rw [inf_eq_half_smul_add_sub_abs_sub' ℝ]
```
and below

File: Mathlib/Topology/ContinuousFunction/Algebra.lean [hunk 4]
```diff
@@ -900,45 +901,27 @@ We now provide formulas for `f ⊓ g` and `f ⊔ g`, where `f g : C(α, β)`,
...
 
 section Lattice
 
 variable {α : Type _} [TopologicalSpace α]
 
-variable {β : Type _} [LinearOrderedField β] [TopologicalSpace β] [OrderTopology β]
-  [TopologicalRing β]
+variable {β : Type _} [TopologicalSpace β]
+
+/-! C(α, β) is a lattice ordered group -/
```
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Algebra.lean:912]:
```suggestion
/-! `C(α, β)`is a lattice ordered group -/
```

**Review by eric-wieser — APPROVED** [review]
LGTM

@semorrison, I think you might have written the `inf_eq` lemma; does this look good to you?

File: Mathlib/Algebra/Order/LatticeGroup.lean (hunk 2 above)
Comment (mans0954) [reply Mathlib/Algebra/Order/LatticeGroup.lean:64]: Possibly it's some odd artefact of the porting process? I think it used to be here: https://github.com/leanprover-community/mathlib/blob/master/src/algebra/order/lattice_group.lean#L6

Comment (mans0954) [reply Mathlib/Algebra/Order/LatticeGroup.lean:64]: Squares being used [here](https://github.com/leanprover-community/mathlib4/commit/4dced5ef925370e6ddef036201dbdb0be41f0520#diff-7f541e62bd43db13b165b8cba1216788fae6d5b2ba7fc4184af231ec348c5ac4R415) and [here](https://github.com/leanprover-community/mathlib4/commit/4dced5ef925370e6ddef036201dbdb0be41f0520#diff-7f541e62bd43db13b165b8cba1216788fae6d5b2ba7fc4184af231ec348c5ac4R425)

Comment (mans0954) [reply Mathlib/Algebra/Order/LatticeGroup.lean]: It no longer seems to be necessary to explicitly import `Mathlib.Algebra.GroupPower.Basic` so I've removed both it and the comment.

File: Mathlib/Algebra/Order/LatticeGroup.lean (hunk 1 above)
Comment (mans0954) [reply Mathlib/Algebra/Order/LatticeGroup.lean]: I've removed the instance, and the rest of the PR still works.

**Comment by j-loreaux** [comment]
This looks fine to me also. The `inf_eq` and `sup_eq` lemmas are easily obtained from the more general versions now, so I don't see any reason not to merge as is.

bors merge

**Pull request closed without merging**